datafusion/expr-common/src/lib.rs

```diff
@@ -17,9 +17,13 @@
 
 //! Logical Expr types and traits for [DataFusion]
 //!
-//! This crate contains types and traits that are used by both Logical and Physical expressions.
-//! They are kept in their own crate to avoid physical expressions depending on logical expressions.
-//!  
+//! This crate contains types and traits that are used by both Logical and
+//! Physical expressions. They are kept in their own crate to avoid physical
+//! expressions depending on logical expressions.
+//!
+//! Note this crate is not intended to have substantial logic itself, but rather
```

Comment:
I added some additional explanation about this crate (partly as a justification about why it was ok to depend on in the physical optimizer)